src/TabContainer.js

```diff
@@ -0,0 +1,89 @@
+import React, { PropTypes } from 'react';
+import uncontrollable from 'uncontrollable';
+
+let idPropType = PropTypes.oneOfType([
+  PropTypes.string,
+  PropTypes.number
+]);
+
+
+let TabContainer = React.createClass({
+
+  propTypes: {
+    /**
+     * HTML id attribute, required if no `generateChildId` prop
+     * is specified.
+     */
+    id(props, ...args) {
+      let error = null;
+
+      if (!props.generateChildId) {
+        error = idPropType(props, ...args);
+
+        if (!error && !props.id) {
+          error = new Error('You must provide an either an `id` or a `generateChildId` prop to TabContainer');
```

Comment:
Is this a hard requirement now, or only for a11y? If the latter, might be worth noting.